stock_report_quantity_by_location/wizards/stock_report_quantity_by_location_prepare.py

```diff
@@ -79,3 +84,40 @@ def _compute_stock_report_by_location(self):
                         }
                     )
         self.env["stock.report.quantity.by.location"].create(vals_list)
+
+    def button_export_html(self):
+        self.ensure_one()
+        action = self.env.ref(
+            "stock_report_quantity_by_location."
+            "action_stock_report_quantity_by_location_html"
+        )
+        vals = action.read()[0]
```

Comment:
**issue**: we should use the proper way to open an action

```suggestion
        action = self.env["ir.actions.act_window"]._for_xml_id(
            "stock_report_quantity_by_location."
            "action_stock_report_quantity_by_location_html"
        )
```

Following this change, rename the following instances of `vals` with `action`

Reply:
Used "ir.actions.actions" to get an action